statements.jl

```diff
@@ -0,0 +1,15 @@
+#=
+This file was manually genererated by executing:
+
+    julia --project=. -Jmoibenchmark.so --trace-compile=stderr pmedian.jl
+
+The procompile statements are written in the screen. They were copied and
+pasted to this file.
+=#
```

Comment:
Maybe add why this is needed? The prints and shows shouldn't really be needed? Does this file actually make any extra difference?

Reply:
there was a difference here but maybe is just my limited memory kicking in, maybe you can test in the large machine.
You can either leave this file blank or remove this line:
`precompile_statements_file="statements.jl"`
from `precompile.jl`

Reply:
If I run the `--trace-compile`, I get this
```
precompile(Tuple{typeof(Base._compute_eltype), Type{Tuple{Base.Pair{String, getfield(Benchmark, Symbol("#31#34"))}, Base.Pair{String, getfield(Benchmark, Symbol("#32#35"))}, Base.Pair{String, getfield(Benchmark, Symbol("#33#36"))}}}})
precompile(Tuple{typeof(Benchmark.run_paper_benchmark), Array{Int64, 1}})
precompile(Tuple{getfield(JSON.Parser, Symbol("##parse#1")), Type{T} where T, Type{Int64}, Bool, Nothing, typeof(JSON.Parser.parse), String})
precompile(Tuple{typeof(JSON.Parser.parse_value), JSON.Parser.ParserContext{Base.Dict{String, Any}, Int64, true, nothing}, JSON.Parser.MemoryParserState})
precompile(Tuple{typeof(Base.setindex!), Base.Dict{String, Any}, Float64, String})
precompile(Tuple{typeof(Base.empty), Base.Dict{Any, Any}, Type{String}, Type{getfield(Benchmark, Symbol("#31#34"))}})
precompile(Tuple{typeof(Base.setindex!), Base.Dict{String, getfield(Benchmark, Symbol("#31#34"))}, Function, String})
precompile(Tuple{getfield(Mmap, Symbol("#3#5")){Int64, Ptr{Nothing}}, Array{UInt8, 1}})
precompile(Tuple{typeof(Base.grow_to!), Base.Dict{String, getfield(Benchmark, Symbol("#31#34"))}, Tuple{Base.Pair{String, getfield(Benchmark, Symbol("#31#34"))}, Base.Pair{String, getfield(Benchmark, Symbol("#32#35"))}, Base.Pair{String, getfield(Benchmark, Symbol("#33#36"))}}, Int64})
precompile(Tuple{typeof(Base.empty), Base.Dict{String, getfield(Benchmark, Symbol("#31#34"))}, Type{String}, Type{Function}})
precompile(Tuple{typeof(Base.merge!), Base.Dict{String, Function}, Base.Dict{String, getfield(Benchmark, Symbol("#31#34"))}})
precompile(Tuple{typeof(Base.setindex!), Base.Dict{String, Function}, Function, String})
precompile(Tuple{typeof(Base.grow_to!), Base.Dict{String, Function}, Tuple{Base.Pair{String, getfield(Benchmark, Symbol("#31#34"))}, Base.Pair{String, getfield(Benchmark, Symbol("#32#35"))}, Base.Pair{String, getfield(Benchmark, Symbol("#33#36"))}}, Int64})
precompile(Tuple{Base.Fix2{typeof(Base.isequal), DataType}, Type{T} where T})
precompile(Tuple{Base.Fix2{typeof(Base.isequal), UnionAll}, Type{T} where T})
precompile(Tuple{typeof(MathOptInterface.Bridges.Variable.concrete_bridge_type), DataType, Type{MathOptInterface.Nonpositives}})
precompile(Tuple{typeof(Base.Broadcast.broadcasted), Function, Function, Array{MathOptInterface.ScalarAffineTerm{Float64}, 1}})
precompile(Tuple{typeof(Base.println), Base.TTY, String, Vararg{String, N} where N})
```